qt_ui/windows/basemenu/QBaseMenu2.py

```diff
@@ -170,3 +170,10 @@ def get_base_image(self):
 
     def new_package(self) -> None:
         Dialog.open_new_package_dialog(self.cp, parent=self.window())
+
+    def update_dialogue_budget(self, budget):
```

Comment:
Missing typehint here.  Typing isn't universally present in these QT modules due to mypy not playing nice with the QT stuff, but we should specify that `budget` param is `int`. 

Otherwise this all looks great to me, thank you!

Reply:
np I'll add it in